src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/Common/DbConnectionStringCommon.cs

```diff
@@ -214,6 +214,136 @@ internal static string ColumnEncryptionSettingToString(SqlConnectionColumnEncryp
             }
         }
 
+        #region <<AttestationProtocol Utility>>
+
+        /// <summary>
+        /// Attestation Protocol.
+        /// </summary>
+        const string AttestationProtocolHGS = "HGS";
+        const string AttestationProtocolAAS = "AAS";
+
+        /// <summary>
+        ///  Convert a string value to the corresponding SqlConnectionAttestationProtocol
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="result"></param>
+        /// <returns></returns>
+        internal static bool TryConvertToAttestationProtocol(string value, out SqlConnectionAttestationProtocol result)
+        {
+            bool isSuccess = false;
+
+            if (StringComparer.InvariantCultureIgnoreCase.Equals(value, AttestationProtocolHGS))
+            {
+                result = SqlConnectionAttestationProtocol.HGS;
+                isSuccess = true;
+            }
+            else if (StringComparer.InvariantCultureIgnoreCase.Equals(value, AttestationProtocolAAS))
+            {
+                result = SqlConnectionAttestationProtocol.AAS;
+                isSuccess = true;
+            }
+            else
+            {
+                result = DbConnectionStringDefaults.AttestationProtocol;
+            }
+
+            return isSuccess;
+        }
+
+        internal static bool IsValidAttestationProtocol(SqlConnectionAttestationProtocol value)
+        {
+            Debug.Assert(Enum.GetNames(typeof(SqlConnectionAttestationProtocol)).Length == 3, "SqlConnectionAttestationProtocol enum has changed, update needed");
+            return value == SqlConnectionAttestationProtocol.NotSpecified 
+                || value == SqlConnectionAttestationProtocol.HGS 
+                || value == SqlConnectionAttestationProtocol.AAS;
+
+        }
+
+        internal static string AttestationProtocolToString(SqlConnectionAttestationProtocol value)
+        {
+            Debug.Assert(IsValidAttestationProtocol(value), "value is not a valid attestation protocol");
+
+            switch (value)
+            {
+                case SqlConnectionAttestationProtocol.HGS:
+                    return AttestationProtocolHGS;
+                case SqlConnectionAttestationProtocol.AAS:
+                    return AttestationProtocolAAS;
+                default:
+                    return null;
+            }
+        }
+
+        internal static SqlConnectionAttestationProtocol ConvertToAttestationProtocol(string keyword, object value)
+        {
+            if (null == value)
+            {
+                return DbConnectionStringDefaults.AttestationProtocol;
+            }
+
+            string sValue = (value as string);
+            SqlConnectionAttestationProtocol result;
+
+            if (null != sValue)
+            {
+                if (TryConvertToAttestationProtocol(sValue, out result))
```

Comment:
Let's remove this and only try once after trimming the spaces

Reply:
Changed